load tests fail obscurely when nix dependencies change

The `Loadtest (Nix)` CI check compares Postgrest versions on the `main` and PR branches. If the nix dependencies are changed in a PR, this will typically cause build failures of the `main` branch, causing the check to fail.

E.g., in PRs #2306 and #2292, we've seen the `Loadtest (Nix)` CI check failing with

```
Running loadtest on "main"...
2022/06/07 23:54:11 encode: can't detect encoding of "stdin"
Error: Process completed with exit code 1.
```

(e.g. here: https://github.com/PostgREST/postgrest/runs/6784583539?check_suite_focus=true)

- [x] improve the test output so that it's clear what's happening #2309
- [ ] skip the CI check or make it optional when the cabal file changes
- [ ] (stretch goal) fix the load test to work even when dependencies change

--- a/nix/tools/withTools.nix
+++ b/nix/tools/withTools.nix
@@ -244,9 +244,9 @@ let
       ''
         export PGRST_SERVER_UNIX_SOCKET="$tmpdir"/postgrest.socket
 
+        rm -f result
         echo -n "Building postgrest... "
-        ${cabal-install}/bin/cabal v2-build ${devCabalOptions} > "$tmpdir"/build.log 2>&1 \
-          || {
+        nix-build -A postgrestPackage > "$tmpdir"/build.log 2>&1 || {
           echo "failed, output:"
           cat "$tmpdir"/build.log
           exit 1
@@ -254,12 +254,8 @@ let
         echo "done."
 
         echo -n "Starting postgrest... "
-        ${cabal-install}/bin/cabal v2-run ${devCabalOptions} --verbose=0 -- \
-          postgrest ${legacyConfig} > "$tmpdir"/run.log 2>&1 &
-
-        # to get the pid of the postgrest process, we need to jump through some hoops
-        # $! will return the pid of cabal - but killing this, will not propagate to postgrest
-        pid=$(timeout -s TERM 1 ${waitForPgrstPid})
+        ./result/bin/postgrest ${legacyConfig} > "$tmpdir"/run.log 2>&1 &
+        pid=$!
         cleanup() {
           kill "$pid" || true
         }

